sdk/src/Core/Amazon.Runtime/Pipeline/HttpHandler/_netstandard/HttpRequestMessageFactory.cs

```diff
@@ -483,6 +483,42 @@ public HttpContent GetRequestContent()
         /// Returns the HTTP response.
         /// </summary>
         /// <returns>The HTTP response.</returns>
+#if NET8_0_OR_GREATER
+        public IWebResponseData GetResponse()
+        {
+            try
+            {
+                var responseMessage = _httpClient.Send(_request, HttpCompletionOption.ResponseHeadersRead);
+
+                return ProcessHttpResponseMessage(responseMessage);
+            }
+            catch (HttpRequestException httpException)
+            {
+                if (httpException.InnerException != null)
+                {
+                    if (httpException.InnerException is IOException)
+                    {
+                        throw httpException.InnerException;
+                    }
+
+                    if (httpException.InnerException is WebException)
+                        throw httpException.InnerException;
+                }
+
+                throw;
+            }
+            catch (OperationCanceledException canceledException)
+            {
+                if (canceledException.InnerException != null)
```

Comment:
Does this need to do a check if it isn't an actual cancellation request before pulling out the inner exception? Example from below: https://github.com/aws/aws-sdk-net/pull/3076/files#diff-2c03c359ab65379b0eb91fd4fb6e2f7833cf3a0a380dacce9a053f8bd2d4b4a0R575

My concern is unwrapping it and throwing can hide IsCancellationRequest value on the rethrown inner exception when it should never have been unwrapped.

Reply:
There is no `cancellationToken` in the sync code path. In the code you linked before the `GetResponseAsync` takes in a `CancellationToken` but the sync `GetResponse` does not pass in a `CancellationToken`. That is why I removed that code from the catch block. I don't think I have every seen sync APIs take in `CancellationToken`.